### PR TITLE
clean: delete `stub_msvc.h`

### PR DESCRIPTION
**Pull request by shenlebantongying**
`strcasecmp` is not standard C, it is defined by POSIX.

(The actual implementation is just looping through chars and do `tolower` https://github.com/bminor/glibc/blob/master/string/strcasecmp.c )

Replace them with QByteArrayView which doesn't own the memory, and it is pretty feature rich (than std::string_view).  

Let's simply take a "view" and do the comparisons (which invokes lower level C functions, but with a prettier interface).

(I find the result code is quite readable, btw.)

### CODE REVIEW

File: src/dict/dsl.cc [hunk 1]
```diff
@@ -1740,12 +1736,16 @@ vector< sptr< Dictionary::Class > > makeDictionaries( vector< string > const & f
...
 
     int extSize = ( uncompressedDsl ? 4 : 7 );
-    if ( fileName.size() - extSize >= 5
-         && strncasecmp( fileName.c_str() + fileName.size() - extSize - 5, "_abrv", 5 ) == 0 ) {
-      // It is, skip it
-      continue;
+
+    if ( fileName.size() - extSize >= 5 ) {
+      auto fileName_without_extension = QByteArrayView( fileName ).chopped( extSize );
+      if ( fileName_without_extension.endsWith( "_abrv" ) ) {
```
Comment (xiaoyifang) [inline src/dict/dsl.cc:1742]: `fileName_without_extension`   lower case

File: src/langcoder.cc [hunk 2]
```diff
@@ -231,10 +229,10 @@ QString LangCoder::intToCode2( quint32 val )
 
 quint32 LangCoder::findIdForLanguage( gd::wstring const & lang )
 {
-  const auto langFolded = Utf8::encode( lang );
+  const auto langFolded = QByteArrayView( Utf8::encode( lang ) );
 
   for ( auto const & lc : LANG_CODE_MAP ) {
-    if ( strcasecmp( langFolded.c_str(), lc.lang.c_str() ) == 0 ) {
```
Comment (xiaoyifang) [inline src/langcoder.cc:237]: strcasecmp? not strncasecmp

File: src/dict/dsl.cc (hunk 1 above)
Comment (shenlebantongying) [reply src/dict/dsl.cc]: Replaced with `strncasecmp/strnicmp -> qstrnicmp`.

Comment (shenlebantongying) [reply src/dict/dsl.cc]: This is the whole implementation of `qstrnicmp`

https://github.com/qt/qtbase/blob/18ec9c7b62f6f1a57cd544bbd2d3e9187f8cd990/src/corelib/text/qbytearray.cpp#L343-L357

Comment (shenlebantongying) [reply src/dict/dsl.cc]: 😅 is this an improvement?

image

File: src/dict/dsl.cc [hunk 3]
```diff
@@ -1737,11 +1733,10 @@ vector< sptr< Dictionary::Class > > makeDictionaries( vector< string > const & f
...
     }
 
-    // Make sure it's not an abbreviation file
+    // Make sure it's not an abbreviation file. extSize of ".dsl" or ".dsl.dz"
 
-    int extSize = ( uncompressedDsl ? 4 : 7 );
-    if ( fileName.size() - extSize >= 5
-         && strncasecmp( fileName.c_str() + fileName.size() - extSize - 5, "_abrv", 5 ) == 0 ) {
+    if ( int extSize = ( uncompressedDsl ? 4 : 7 ); ( fileName.size() >= ( 5 + extSize ) )
+         && QByteArrayView( fileName ).chopped( extSize ).last( 5 ).compare( "_abrv", Qt::CaseInsensitive ) ) {
```
Comment (xiaoyifang) [inline src/dict/dsl.cc:1739]: can  use Utils::endsWithIgnoreCase

**Comment by shenlebantongying** [comment]
amended by https://github.com/xiaoyifang/goldendict-ng/pull/1971

**Pull request merged**